ui/app/serializers/clients/activity.js

```diff
@@ -29,12 +32,50 @@ export default class ActivitySerializer extends ApplicationSerializer {
     });
   }
 
-  // For 1.10 release naming changed from 'distinct_entities' to 'entity_clients' and
+  // for vault usage - vertical bar chart
+  flattenByMonths(payload, isNewClients = false) {
+    if (isNewClients) {
+      return payload.map((m) => {
+        return {
+          month: m.timestamp,
+          entity_clients: m['new_clients']['counts']['entity_clients'],
+          non_entity_clients: m['new_clients']['counts']['non_entity_clients'],
+          total: m['new_clients']['counts']['clients'],
+          namespaces: this.flattenDataset(m['new_clients']['namespaces']),
+        };
+      });
+    } else {
+      return payload.map((m) => {
+        return {
+          month: m.timestamp,
+          entity_clients: m['counts']['entity_clients'],
+          non_entity_clients: m['counts']['non_entity_clients'],
+          total: m['counts']['clients'],
+          namespaces: this.flattenDataset(m['namespaces']),
+          new_clients: {
+            entity_clients: m['new_clients']['counts']['entity_clients'],
+            non_entity_clients: m['new_clients']['counts']['non_entity_clients'],
+            total: m['new_clients']['counts']['clients'],
+            namespaces: this.flattenDataset(m['new_clients']['namespaces']),
+          },
+        };
+      });
+    }
+  }
+
+  // In 1.10 'distinct_entities' changed to 'entity_clients' and
   // 'non_entity_tokens' to 'non_entity_clients'
-  // accounting for deprecated API keys here and updating to latest nomenclature
   homogenizeClientNaming(object) {
-    // TODO CMB check with API payload, latest draft includes both new and old key names
-    // TODO CMB Delete old key names IF correct ones exist?
+    // if new key names exist, only return those key/value pairs
+    if (Object.keys(object).includes('entity_clients', 'non_entity_clients')) {
```

Comment:
I think `Array.includes` only takes one parameter so this should be updated to an or statement, one for each. Is it reasonable to assume that if we have one of the updated names, we have both? In that case we can maybe just check for one.

Reply:
Yes - it sure does 😅 - great idea thank you!